pkgs/dart_tooling_mcp_server/lib/src/utils/cli_utils.dart

```diff
@@ -102,3 +144,30 @@ Future<CallToolResult> runCommandInRoots(
   }
   return CallToolResult(content: outputs);
 }
+
+/// Returns whether or not [rootUri] is an allowed root, either exactly matching
+/// or under on of the [knownRoots].
+bool _isAllowedRoot(String rootUri, List<Root> knownRoots) =>
+    knownRoots.any((knownRoot) => knownRoot.uri.startsWith(rootUri));
```

Comment:
shouldn't this check that `rootUri.startsWith(knownRoot.uri)` instead?

Reply:
I rewrote this to also make sure you can't circumvent it using `../` and added tests